ObjectMapper/Core/ToJSON.swift

```diff
@@ -11,9 +11,21 @@ import Foundation
 class ToJSON {
 	
     func basicType<N>(field: N, key: String, inout dictionary: [String : AnyObject]) {
-		var temp = dictionary
 		var currentKey = key
 
+		// Nested keys
+		var allKeys = currentKey.componentsSeparatedByString(".")
+		if allKeys.count > 1 {
+			currentKey = allKeys.first!
+			allKeys.removeAtIndex(0)
+			let remainingKey = join(".", allKeys)
```

Comment:
Why don't you using another array value like `join(".", Array(allKeys[1..<allKeys.count]))`, and you can remove `allKeys.removeAtIndex(0)` and use `let` for `allKeys` for more immutability.